### PR TITLE
Fixes bug https://github.com/OmniSharp/omnisharp-roslyn/issues/1225

### PR DESCRIPTION
**Pull request by dmgonch**
One potential controversial issue with the changes is that I tricked git to include bin/obj folders in the new test-assets\test-projects\DotNetCoreAppSigned. If desired I can zip the test files and unzip the project during the test.

### CODE REVIEW

**Comment by dmgonch** [comment]
To clarify the need of the extra files in test assets: the assembly is delay signed and I couldn’t find a good repro without these extra files in obj/bin that Visual Studio generated.

File: src/OmniSharp.MSBuild/ProjectManager.cs [hunk 1]
```diff
@@ -483,22 +483,31 @@ private void UpdateReferences(Project project, ImmutableArray<string> referenceP
...
                 {
                     _logger.LogWarning($"Unable to resolve assembly '{referencePath}'");
+                    continue;
                 }
-                else
+
+                // There is no need to explicitly add assembly to workspace when the assembly is produced by a project reference.
+                // Doing so actually can cause /codecheck request to return errors like below for types in the referenced project if it is for example signed:
+                // The type 'TestClass' exists in both 'SignedLib, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' and 'ClassLibrary1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=a5d85a5baa39a6a6'
+                if (TryFindLoadedProjectReferenceWithTargetPath(referencePath, projectReferencePaths, project.Name, out ProjectFileInfo projectReferenceWithTarget))
```
Comment (DustinCampbell) [inline src/OmniSharp.MSBuild/ProjectManager.cs]: Should we just build up a `HashSet<string>` of project target paths ahead of time to avoid searching the list of project references for each metadata reference?

File: test-assets/test-projects/DotNetStdSigned/CalleeLib/obj/project.assets.json [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  "packageFolders": {
+    "C:\\Users\\dgonch\\.nuget\\packages\\": {},
+    "C:\\Program Files\\dotnet\\sdk\\NuGetFallbackFolder": {}
+  },
+  "project": {
+    "version": "1.0.0",
+    "restore": {
+      "projectUniqueName": "C:\\temp\\DotNetStdSigned\\CalleeLib\\CalleeLib.csproj",
+      "projectName": "CalleeLib",
+      "projectPath": "C:\\temp\\DotNetStdSigned\\CalleeLib\\CalleeLib.csproj",
```
Comment (DustinCampbell) [inline test-assets/test-projects/DotNetStdSigned/CalleeLib/obj/project.assets.json]: Note that `project.assets.json` files contain absolute file paths to your machine. So, they're not actually valid on the test machine. Why are these needed?

**Comment by dmgonch** [comment]
Now that you pointed me to build.json I see that there is no point in checking in the test outputs. Removed. Thanks!.

File: test-assets/test-projects/DotNetStdSigned/CalleeLib/obj/project.assets.json (hunk 2 above)
Comment (dmgonch) [reply test-assets/test-projects/DotNetStdSigned/CalleeLib/obj/project.assets.json]: Indeed after switching to use build.json as per your other comment I don't need test outputs in the sources. Removed.

File: src/OmniSharp.MSBuild/ProjectManager.cs (hunk 1 above)
Comment (dmgonch) [reply src/OmniSharp.MSBuild/ProjectManager.cs]: Given that typically the list of project references is relatively short IMO it is acceptable here to sacrifice a little bit of perf for avoiding managing an additional piece of workspace state, particularly for the cases when the list of project is updated.
Please advise if you still would prefer me to look into adding this optimization.

Comment (DustinCampbell) [reply src/OmniSharp.MSBuild/ProjectManager.cs]: No need to make the change. Just asking.
